scipy/stats/tests/test_distributions.py

```diff
@@ -1799,6 +1799,65 @@ def test_fit_simple_non_random_data(self):
         assert_allclose(loc, 4, atol=1e-15, rtol=1e-15)
 
 
+class TestPowerlaw(object):
+    @pytest.mark.parametrize('a', [.5, 1.5, 5])
+    def test_fit_shape(self, a):
+        data = stats.powerlaw.rvs(size=1000, a=a)
+        # obtain the unpenalized log-likelihood function
+
+        def ll(args, data):
+            # to get the same behavior as `powerlaw.reduce_func`, negate the
+            # log-likelihood function
+            return -np.sum(np.log(stats.powerlaw.pdf(data, *args)), axis=0)
+
+        # compare the results of the LL
+        # a warning is raised when data is generated in some situations, so
+        # it will be supressed.
+        with np.errstate(divide='ignore'):
+            _assert_less_or_close_loglike(stats.powerlaw, data, ll)
+
+    @pytest.mark.parametrize("a", [.5, 1, 1.5])
+    @pytest.mark.parametrize("loc", [0, 1])
+    @pytest.mark.parametrize("scale", [1, 5])
+    @pytest.mark.parametrize("fa", [True, False])
+    @pytest.mark.parametrize("floc", [True, False])
+    @pytest.mark.parametrize("fscale", [True, False])
+    def test_fit_MLE_comp_optimzer_combinations(self, a, loc, scale, fa, floc,
+                                                fscale):
+        if False not in {fa, floc, fscale}:
+            pytest.skip()
+        data = stats.powerlaw.rvs(size=2500, a=a, loc=loc, scale=scale)
+        
+        def ll(args, data):
+            # to get the same behavior as `powerlaw.reduce_func`, negate the
+            # log-likelihood function
+            return -np.sum(np.log(stats.powerlaw.pdf(data, *args)), axis=0)
+
+        # build kwds based on which arguments are indicated to be fixed
+        kwds = dict()
+        if fa:
+            kwds['f0'] = a
+        if floc:
+            kwds['floc'] = loc
+        if fscale:
+            kwds['fscale'] = scale
+        with np.errstate(divide='ignore'):
+            mle_analytical = stats.powerlaw.fit(data, **kwds)
+            numerical_opt = super(type(stats.powerlaw),
+                                  stats.powerlaw).fit(data, **kwds)
+            ll_mle_analytical = ll(mle_analytical, data)
+            ll_numerical_opt = ll(numerical_opt, data)
+            try:
+                assert (ll_mle_analytical <= ll_numerical_opt)
+            except Exception as e1:
+                try:
+                    np.testing.assert_almost_equal(ll_mle_analytical,
+                                                   ll_numerical_opt)
+                except Exception as e2:
+                    np.testing.assert_almost_equal(mle_analytical,
+                                                   numerical_opt, decimal=3)
+
```

Comment:
This could become more succinct, but the objective was to test these things in this order: 
1. that LL of the override is <= the super method: ll_mle_analytical <= ll_numerical_opt
2. that the LLs are almost equal
3. then finally, assuming that those both failed in some way, see if the actual values fitted are close assert_almost_equal(mle_analytical, numerical_opt)